### PR TITLE
[14.0][ADD] stock_truck_unloading: New module

### PR DESCRIPTION
**Pull request by Ricardoalso**
Module to manage truck unloading time

***Result***

### CODE REVIEW

**Comment by rousseldenis** [comment]
@Ricardoalso @jbaudoux Isn't this similar to https://github.com/OCA/stock-logistics-workflow/tree/16.0/stock_grn ?

**Comment by Ricardoalso** [comment]
> @Ricardoalso @jbaudoux Isn't this similar to https://github.com/OCA/stock-logistics-workflow/tree/16.0/stock_grn ?

@rousseldenis Indeed, thanks !

File: stock_truck_unloading/models/stock_picking.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    @api.onchange("manage_truck_arrival")
+    def on_change_manage_truck_arrival(self):
+        if not self.manage_truck_arrival:
+            self.is_arrived = False
+            self.date_truck_arrival = False
+
+    def do_truck_arrived(self):
+        for pick in self:
+            if pick.manage_truck_arrival:
+                pick.is_arrived = not pick.is_arrived
```
Comment (legalsylvain) [inline stock_truck_unloading/models/stock_picking.py:26]: If you can do and undo truck arrived, better to name the function `toggle_truck_arrived`.

File: stock_truck_unloading/views/stock_picking_views.xml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            </xpath>
+        </field>
+    </record>
+
+    <record id="view_picking_form_button_truck_arrived" model="ir.ui.view">
+        <field name="model">stock.picking</field>
+        <field name="inherit_id" ref="stock.view_picking_form"/>
+        <field name="arch" type="xml">
+            <header position="inside">
+                <button name="do_truck_arrived" type="object" string="Truck Arrived" attrs="{'invisible': [('manage_truck_arrival', '=', False)]}" />
```
Comment (legalsylvain) [inline stock_truck_unloading/views/stock_picking_views.xml:23]: same here for the string. it should be "Truck Arrived" or "Truck not Arrived", depending on is_arrived.

File: stock_truck_unloading/models/stock_picking.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+from datetime import datetime
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    manage_truck_arrival = fields.Boolean(default=False)
```
Comment (legalsylvain) [inline stock_truck_unloading/models/stock_picking.py:13]: I guess having a default based on the stock.picking.type is necerrary here. Don't you think ?

**Review by legalsylvain — COMMENTED** [review]
Functional review. Some comments.

Thanks for your contribution.

**Comment by Ricardoalso** [comment]
> Functional review. Some comments.
> 
> Thanks for your contribution.

Thank you for your inputs @legalsylvain 

Closing the PR due to the discovery of stock_grn

**Pull request closed without merging**